src/Hng.Domain/Entities/Product.cs

```diff
@@ -1,9 +1,24 @@
 
+using System.ComponentModel.DataAnnotations.Schema;
+using System.ComponentModel.DataAnnotations;
+
 namespace Hng.Domain.Entities;
 public class Product : EntityBase
 {
+    public Product()
+    {
+        CreatedAt = DateTime.UtcNow;
+    }
+
+    [Key]
+    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+    public new int Id { get; set; }
     public string Name { get; set; }
     public string Description { get; set; }
+    public string[] Category { get; set; }
```

Comment:
Having a prop with as an array can be tricky in my opinion. Create another entity for a category, properly specifying the relationship will be better.

Reply:
Done